packaging/installer/kickstart-ng.sh

```diff
@@ -0,0 +1,842 @@
+                             write files to the temporary directory, and run files from that location.
+  ROOTCMD:                   Used to specify a command to use to run another command with root privileges if needed. By
+                             default we try to use sudo, doas, or pkexec (in that order of preference), but if
+                             you need special options for one of those to work, or have a different tool to do
+                             the same thing on your system, you can specify it here.
+  DO_NOT_TRACK               If set to a value other than 0, behave as if \`--disable-telemetry\` was specified.
+
+HEREDOC
+}
+
+# ======================================================================
+# Utility functions
+
+setup_terminal() {
+  TPUT_RESET=""
+  TPUT_WHITE=""
+  TPUT_BGRED=""
+  TPUT_BGGREEN=""
+  TPUT_BOLD=""
+  TPUT_DIM=""
+
+  # Is stderr on the terminal? If not, then fail
+  test -t 2 || return 1
+
+  if command -v tput > /dev/null 2>&1; then
+    if [ $(($(tput colors 2> /dev/null))) -ge 8 ]; then
+      # Enable colors
+      TPUT_RESET="$(tput sgr 0)"
+      TPUT_WHITE="$(tput setaf 7)"
+      TPUT_BGRED="$(tput setab 1)"
+      TPUT_BGGREEN="$(tput setab 2)"
+      TPUT_BOLD="$(tput bold)"
+      TPUT_DIM="$(tput dim)"
+    fi
+  fi
+
+  echo "${TPUT_RESET}"
+
+  return 0
+}
+
+fatal() {
+  printf >&2 "%s\n\n" "${TPUT_BGRED}${TPUT_WHITE}${TPUT_BOLD} ABORTED ${TPUT_RESET} ${*}"
+  exit 1
+}
+
+run_ok() {
+  printf >&2 "%s\n\n" "${TPUT_BGGREEN}${TPUT_WHITE}${TPUT_BOLD} OK ${TPUT_RESET}"
+}
+
+run_failed() {
+  printf >&2 "%s\n\n" "${TPUT_BGRED}${TPUT_WHITE}${TPUT_BOLD} FAILED ${TPUT_RESET}"
+}
+
+ESCAPED_PRINT_METHOD=
+# shellcheck disable=SC3050
+if printf "%q " test > /dev/null 2>&1; then
+  ESCAPED_PRINT_METHOD="printfq"
+fi
+
+escaped_print() {
+  if [ "${ESCAPED_PRINT_METHOD}" = "printfq" ]; then
+    # shellcheck disable=SC3050
+    printf "%q " "${@}"
+  else
+    printf "%s" "${*}"
+  fi
+  return 0
+}
+
+progress() {
+  echo >&2 " --- ${TPUT_BOLD}${*}${TPUT_RESET} --- "
+}
+
+run_logfile="/dev/null"
+run() {
+  user="${USER--}"
+  dir="${PWD}"
+
+  if [ "$(id -u)" = "0" ]; then
+    info="[root ${dir}]# "
+    info_console="[${TPUT_DIM}${dir}${TPUT_RESET}]# "
+  else
+    info="[${user} ${dir}]$ "
+    info_console="[${TPUT_DIM}${dir}${TPUT_RESET}]$ "
+  fi
+
+  {
+    printf "%s" "${info}"
+    escaped_print "${@}"
+    printf " ... "
+  } >> "${run_logfile}"
+
+  printf >&2 "%s" "${info_console}${TPUT_BOLD}"
+  escaped_print >&2 "${@}"
+  printf >&2 "%s\n" "${TPUT_RESET}"
+
+  "${@}"
+
+  ret=$?
+  if [ ${ret} -ne 0 ]; then
+    run_failed
+    printf "%s\n" "FAILED with exit code ${ret}" >> "${run_logfile}"
+  else
+    run_ok
+    printf "OK\n" >> "${run_logfile}"
+  fi
+
+  return ${ret}
+}
+
+warning() {
+  printf >&2 "%s\n\n" "${TPUT_BGRED}${TPUT_WHITE}${TPUT_BOLD} WARNING ${TPUT_RESET} ${*}"
```

Comment:
Small suggestion, since we already use `TPUT_BGRED` for `ABORTED` should we use a lighter background maybe for `WARNING` ?

Reply:
We had been using colored backgrounds, but those have caused problems in the past. Arguably, we shouldn’t even be explicitly setting this as white for the foreground.

Reply:
OK, clarifying further, I think there’s no issue sharing the coloring for aborts and warnings, but I think we should actually completely get rid of the color coding since it’s too dependent on the terminal settings.